src/Microsoft.Identity.Web/MicrosoftIdentityCircuitHandler.cs

```diff
@@ -17,257 +17,257 @@
-            }
-        }
+				ChallengeUser(
+					scopes.ToArray(),
+					claims,
+					userflow);
+			}
+			else
+			{
+				throw exception;
+			}
+		}
 
-        /// <summary>
-        /// Forces the user to consent to specific scopes and perform
-        /// Conditional Access to get specific claims. Use on a Razor/Blazor
-        /// page or controller to proactively ensure the scopes and/or claims
-        /// before acquiring a token. The other mechanism <see cref="HandleException(Exception)"/>
-        /// ensures claims and scopes requested by Azure AD after a failed token acquisition attempt.
-        /// See https://aka.ms/ms-id-web/ca_incremental-consent for details.
-        /// </summary>
-        /// <param name="scopes">Scopes to request.</param>
-        /// <param name="claims">Claims to ensure.</param>
-        /// <param name="userflow">Userflow being invoked for AAD B2C.</param>
-        public void ChallengeUser(
-            string[]? scopes,
-            string? claims = null,
-            string? userflow = null)
-        {
-            IEnumerable<string> effectiveScopes = scopes ?? new string[0];
+		/// <summary>
+		/// Forces the user to consent to specific scopes and perform
+		/// Conditional Access to get specific claims. Use on a Razor/Blazor
+		/// page or controller to proactively ensure the scopes and/or claims
+		/// before acquiring a token. The other mechanism <see cref="HandleException(Exception)"/>
+		/// ensures claims and scopes requested by Azure AD after a failed token acquisition attempt.
+		/// See https://aka.ms/ms-id-web/ca_incremental-consent for details.
+		/// </summary>
+		/// <param name="scopes">Scopes to request.</param>
+		/// <param name="claims">Claims to ensure.</param>
+		/// <param name="userflow">Userflow being invoked for AAD B2C.</param>
+		public void ChallengeUser(
+			string[]? scopes,
+			string? claims = null,
+			string? userflow = null)
+		{
+			IEnumerable<string> effectiveScopes = scopes ?? new string[0];
 
-            string[] additionalBuiltInScopes =
-            {
-                 OidcConstants.ScopeOpenId,
-                 OidcConstants.ScopeOfflineAccess,
-                 OidcConstants.ScopeProfile,
-            };
+			string[] additionalBuiltInScopes =
+			{
+				 OidcConstants.ScopeOpenId,
+				 OidcConstants.ScopeOfflineAccess,
+				 OidcConstants.ScopeProfile,
+			};
 
-            effectiveScopes = effectiveScopes.Union(additionalBuiltInScopes).ToArray();
+			effectiveScopes = effectiveScopes.Union(additionalBuiltInScopes).ToArray();
 
-            string redirectUri;
-            if (IsBlazorServer)
-            {
-                redirectUri = NavigationManager.Uri;
-            }
-            else
-            {
+			string redirectUri;
+			if (IsBlazorServer)
+			{
+				redirectUri = NavigationManager.Uri;
+			}
+			else
+			{
 #pragma warning disable CS8602 // Dereference of a possibly null reference. HttpContext will not be null in this case.
```

Comment:
HttpContext can definitely be null here if the library is attempted to be used within a worker service. With that in mind, the warning should be addressed.

Reply:
this is only for blazorwasm scenarios, so we will always have the httpContext.